civis/_utils.py

```diff
@@ -129,3 +129,19 @@ def f_retry(*args, **kwargs):
         return f_retry
 
     return deco_retry
+
+
+class BufferedPartialReader(object):
+    def __init__(self, buf, max_bytes):
+        self.max_bytes = max_bytes
+        self.bytes_read = 0
+        self.buf = buf
+        self.len = max_bytes
+
+    def read(self, size=-1):
```

Comment:
This defaults `size` to `-1`, but doesn't seem to handle the size=-1 case. `size=-1` should set `size` to `bytes_left`, right?

Reply:
good catch; fixed